file_writer_test.go

```diff
@@ -14,26 +14,21 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-const abcException = "org.apache.hadoop.hdfs.protocol.AlreadyBeingCreatedException"
-
 func appendIgnoreABC(t *testing.T, client *Client, path string) (*FileWriter, error) {
 	// This represents a bug in the HDFS append implementation, as far as I can
 	// tell. Try a few times again, then skip the test.
 	retries := 0
 	for {
 		fw, err := client.Append(path)
 
-		if pathErr, ok := err.(*os.PathError); ok {
-			if nnErr, ok := pathErr.Err.(Error); ok && nnErr.Exception() == abcException {
-				t.Log("Ignoring AlreadyBeingCreatedException from append")
-
-				if retries < 3 {
-					retries += 1
-					continue
-				} else {
-					t.Skip("skipping Append test because of repeated AlreadyBeingCreatedException")
-					return fw, nil
-				}
+		if os.IsExist(err) {
```

Comment:
I think this should stay the way it is (and only ignore the ABC error, rather than any IsExist).

Reply:
`AlreadyBeingCreatedException` was interpreted as `os.ErrExist` for all HDFS requests, so it was impossible to ignore only this exception here. Actually, receiving `os.ErrExist` error on `append` operation is a bit strange, so I decided to change implementation and interpret `AlreadyBeingCreatedException` only on `create`.